design/replace-CLI-with-SDK.md

```diff
@@ -0,0 +1,220 @@
+
+<!-- toc -->
+- [Motivation](#motivation)
+- [Scope](#scope)
+- [High Level Design](#high-level-design)
+  - [Kopia SDK wrappers](#kopia-sdk-wrappers)
+    - [Storage pkg](#storage-pkg)
+    - [Repository pkg](#repository-pkg)
+  - [Repository server controller changes](#repository-server-controller-changes)
+	- [Kopia CLI Approach](#kopia-cli-approach)
+	- [Kopia SDK Approach](#kopia-sdk-approach)
+<!-- /toc -->
+
+
+# Motivation
+
+Kanister uses a kubernetes custom controller makes use of kopia as a primary backup and restore tool. 
+The detailed design of the customer controller can be found [here](https://github.com/kanisterio/kanister/blob/master/design/kanister-kopia-integration.md)
+
+The custom controller called as repository server controller currently uses kopia CLI to perform the kopia operations. All the
```

Comment:
```suggestion
The custom controller called as repository server controller currently uses Kopia CLI to perform the Kopia operations. All the
```
K in Kopia, Kanister and Kubernetes should be in uppercase to maintain branding style